cal-itp/benefits
# Feat: automatically associate user to agency

Closes #2243 

This PR makes it so that users are added to their transit agency's `staff_group` upon login. This is done with the help of a field called `sso_domain` on the transit agency.

The user needs to be manually added to their transit agency's `customer_service_group` to see the part of the dashboard for "In-person enrollment".

## Testing locally

- Set your `GOOGLE_SSO_STAFF_LIST` environment variable to `*`
- Make sure your `GOOGLE_SSO_ALLOWABLE_DOMAINS` environment variable is `compiler.la`

Those two steps make it so your Google account will be seen as a "transit agency staff user" - i.e. not put into the "Cal-ITP" group but still marked as `is_staff`.

- Set `SSO_SHOW_FORM_ON_ADMIN_PAGE` to `True` temporarily in `benefits/settings.py`
- Reset your database to use local fixtures - `./bin/reset_db.sh`
- Set up the SSO domain mapping
   - Launch the app and go to `/admin` and login as your superuser (based on your [`DJANGO_SUPERUSER_USERNAME`](https://docs.calitp.org/benefits/configuration/environment-variables/#django_superuser_username) and `DJANGO_SUPERUSER_PASSWORD` environment variables)
   - Edit the "CST" `TransitAgency` so that it has an `sso_domain` of `compiler.la`
   - Create a new group and set it as the `staff_group` for CST
   - Create another new group and set it as the `customer_service_group` for CST
- Log out
- Log back in using your compiler.la Google account

**Expected result:** your user has been added to the `staff_group` for CST upon login. When you go to the agency dashboard, you see the transit agency name, but no button or anything for in-person enrollment.

To elevate the user to having permission to do in-person enrollment:
- Log in as the superuser
- Add your Google user to the CST customer service group
- Log back in as your Google user

**Expected result:** When you go to the agency dashboard, you see the the transit agency name as well as the in-person enrollment section.

## Screenshots
User in CST's `staff_group` but not the `customer_service_group`

<img src="https://github.com/user-attachments/assets/c33828ae-768e-4c3b-b399-c32a4b0b2387" width="450">

User in CST's `staff_group` and `customer_service_group`

<img src="https://github.com/user-attachments/assets/d844295d-b29b-4d51-a706-c657f2572b5a" width="450">
